src/edu/solvd/mentoring/Support.java

```diff
@@ -0,0 +1,417 @@
+package edu.solvd.mentoring;
+
+import edu.solvd.mentoring.car.Car;
+import edu.solvd.mentoring.car.detail.CarBase;
+import edu.solvd.mentoring.car.detail.Engine;
+import edu.solvd.mentoring.car.detail.Wheel;
+import edu.solvd.mentoring.enums.DetailQuality;
+import edu.solvd.mentoring.road.*;
+import edu.solvd.mentoring.road.surface.Surface;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Random;
+import java.util.Set;
+
+class Support {
+  // make it random so for every new Support() different values are generated
+  //
+  //4 surfaces
+  //
+  //5 wheels
+  //5 engines
+  //3 carBases
+  //
+  //5 cars
+  //3 routes
+  //1 garage
+  private static HashMap<Integer, Surface> surfaces = new HashMap<>();
+
+  private static HashMap<Integer, Wheel> wheels = new HashMap<>();
+  private static HashMap<Integer, Engine> engines = new HashMap<>();
+  private static HashMap<Integer, CarBase> carBases = new HashMap<>();
+
+  private static HashMap<Integer, Car> cars = new HashMap<>();
+  private static HashMap<Integer, Route> routes = new HashMap<>();
+  private static Garage garage = null;
+
+  //just a quick converter
+  private static short s(int i){
+    return (short)i;
+  }
+  private static short s(double d){ return (short)d; }
+
+  //some additional Functions
+  private static short getShort(short maxValue, short minValue){
+    return (short)(Math.round(Math.random() * (maxValue - minValue)) + minValue);
+  }
+  private static HashMap<Integer,Short> getGeneratedClutch(short maxClutch, short minClutch, HashMap<Integer,Surface> surfaces){
+    HashMap<Integer,Short> clutch = new HashMap<>();
+    Set<Integer> surfaceKeys = surfaces.keySet();
+
+    for(Integer i: surfaceKeys){
+      short val = s(getShort(maxClutch,minClutch));
+      clutch.put(i,val);
+    }
+
+    return clutch;
+  }
+
+  private static void initSurfaces(){
+      surfaces.put(1,new Surface("highway",'=',1));
+      surfaces.put(2,new Surface("city road",'-',2));
+      surfaces.put(3,new Surface("ferry road",'+',3));
+      surfaces.put(4,new Surface("country road",'*',4));
+  }
+
+  public static Surface getSurface(int id){
+    if (surfaces.isEmpty()){
+      initSurfaces();
+    }
+    return surfaces.get(id);
+  }
+
+  public static HashMap<Integer, Surface> getSurfaces(){
+    if (surfaces.isEmpty()){
+      initSurfaces();
+    }
+    return surfaces;
+  }
+
+  private static void initWheels(){
+      HashMap<Integer,Short> wheelClutch1 = new HashMap<>();
```

Comment:
Map<Integer,Short> wheelClutch1 = new HashMap<>()

Reply:
Is there any difference?